postreise/plot/plot_interconnection_map.py

```diff
@@ -1,249 +1,179 @@
-        {
-            "xs": branch_mdc1[["from_x", "to_x"]].values.tolist(),
-            "ys": branch_mdc1[["from_y", "to_y"]].values.tolist(),
-            "capacity": branch_mdc1.Pmax.astype(float) * line_width_const * hvdc_width
-            + 0.1,
-            "cap": branch_mdc1.Pmax.astype(float),
-        }
-    )
-
-    # pseudolines for ac
-    multi_line_source5 = ColumnDataSource(
-        {
-            "xs": b2b[["from_x", "to_x"]].values.tolist(),
-            "ys": b2b[["from_y", "to_y"]].values.tolist(),
-            "capacity": b2b.Pmax.astype(float) * 0.00023 + 0.2,
-            "cap": b2b.Pmax.astype(float),
-            "col": (
-                "#006ff9",
-                "#006ff9",
-                "#006ff9",
-                "#006ff9",
-                "#006ff9",
-                "#006ff9",
-                "#006ff9",
-                "#8B36FF",
-                "#8B36FF",
-            ),
-        }
-    )
-
-    # lower 48 states, patches
-    source = ColumnDataSource(
-        dict(
-            xs=a,
-            ys=b,
-            col=["gray" for i in range(48)],
-            col2=["gray" for i in range(48)],
-            label=list(state_counts["count"]),
-            state_name=list(state_counts["state"]),
-        )
-    )
-
-    # Set up figure
-    tools: str = "pan, wheel_zoom, reset, save"
-
     p = figure(
-        tools=tools,
+        tools="pan,wheel_zoom,reset,save",
         x_axis_location=None,
         y_axis_location=None,
-        plot_width=800,
-        plot_height=800,
+        plot_width=figsize[0],
+        plot_height=figsize[1],
         output_backend="webgl",
-        sizing_mode="stretch_both",
+        sizing_mode="scale_both",
         match_aspect=True,
     )
 
-    # for legend, hidden lines
-    leg_clr = ["#006ff9", "#8B36FF", "#01D4ED", "#FF2370"]
-    leg_lab = ["Western", "Eastern", "Texas", "HVDC"]
-    leg_xs = [-1.084288e07] * 4
-    leg_ys = [4.639031e06] * 4
-
-    for (colr, leg, x, y) in zip(leg_clr, leg_lab, leg_xs, leg_ys):
-        p.line(x, y, color=colr, width=5, legend=leg)
+    p.xgrid.visible = False
+    p.ygrid.visible = False
 
-    # pseudo lines for hover tips
-    lines = p.multi_line(
-        "xs", "ys", color="black", line_width="capacity", source=multi_line_source6
+    state_counts = count_nodes_per_state(grid)
+    plot_states(
+        bokeh_figure=p,
+        background_map=background_map,
+        state_list=list(state_counts["state"]),
+        labels_list=list(state_counts["count"]),
+        tooltip_name="nodes",
     )
 
-    # background tiles
-    p.add_tile(get_provider(Vendors.CARTODBPOSITRON))
-
-    # state borders
-    patch = p.patches("xs", "ys", fill_alpha=0.0, line_color="col", source=source)
-
-    # branches
-    source_list = [multi_line_source, multi_line_source2, multi_line_source3]
+    p.multi_line(
+        branch_west[["from_x", "to_x"]].values.tolist(),
+        branch_west[["from_y", "to_y"]].values.tolist(),
```

Comment:
I think `.to_numpy()` is the recommended replacement for `.values`, to make it clearer that what's returned is a copy of the data, not a reference to the data itself.
https://pandas.pydata.org/pandas-docs/stable/whatsnew/v0.24.0.html#accessing-the-values-in-a-series-or-index

Reply:
Done